src/components/ui/DataGrid/DataGrid.tsx

```diff
@@ -55,13 +56,24 @@ export function DataGrid<T>({
   compact = false,
   fitToGridWidth = false,
   loading = true,
+  showQuickFilter = false,
   className = "",
   gridProps = {},
   error,
 }: DataGridProps<T>) {
   const gridRef = useRef<AgGridReact>(null);
   const [gridReady, setGridReady] = useState(false);
 
+  useEffect(() => {
```

Comment:
Instead of using an effect could you instead just pass in `quickFilterText={gridProps.quickFilterText}` to ag-grid? Seems like that may be the currently supported method.